dms/models/directory.py

```diff
@@ -365,25 +388,12 @@ def _search_panel_directory(self, **kwargs):
     # Search
     # ----------------------------------------------------------
     @api.model
-    def _search(
-        self,
-        args,
-        offset=0,
-        limit=None,
-        order=None,
-        count=False,
-        access_rights_uid=None,
-    ):
-        result = super(DmsDirectory, self)._search(
-            args, offset, limit, order, False, access_rights_uid
-        )
-        if result:
-            directory_ids = set(result)
-            if self.env.user.has_group("base.group_portal"):
-                exclude_ids = self._get_ids_without_access_groups("read")
-                directory_ids -= set(exclude_ids)
-                return directory_ids
-        return result
+    def search(self, args, offset=0, limit=None, order=None, count=False):
+        res = super(DmsDirectory, self).search(args, offset, limit, order, False)
+        for item in res:
+            if not item.check_access("read"):
+                res -= item
+        return res
```

Comment:
It seems like this implementation will introduce bugs.

For example, if `count=True`, you're not passing that to `super()`. Also, even if you pass it, it wouldn't return accurate results.

Also, if `offset` or `limit` are used, the returned recordset won't have the expected size.

So, you'd be safer extending `_apply_ir_rules()` directly, which filters search results at SQL level and doesn't have these problems. 🤷🏼‍♂️ 

If you don't want to go that deep, you'll have to do more searches, which would be much less performant, but would work too.

Reply:
I understand that using this is not a good solution and needs to be improved.
To extend `_apply_ir_rules()` and include the accessible `directory_ids` in the query, you would need to do `.search()` of all records with `[('res_model', '! =', False)]` and get all that comply with `check_access("read")` BUT without going into a recursive loop, so a context variable would be used.
Do you think this change would be the correct one?

Reply:
I have changed about it.